include/mbgl/storage/default_file_source.hpp

```diff
@@ -140,10 +141,14 @@ class DefaultFileSource : public FileSource {
     class Impl;
 
 private:
+    // Shared so destruction is done on this thread
```

Comment:
@ivovandongen  Is it possible to clarify why wrapping a pointer in shared_ptr ensure destruction on a specific thread?